Content.Server/GameObjects/Components/Items/DiceComponent.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using Content.Server.GameObjects.Components.Sound;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.Audio;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Maths;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Items
+{
+    public class DiceComponent : Component, IActivate, IUse, ILand, IExamine
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IPrototypeManager _prototypeManager;
+#pragma warning restore 649
+
+        public override string Name => "Dice";
+
+        private Random _random;
+        private int _step = 1;
+        private int _sides = 20;
+        private int _currentSide = 20;
+        [ViewVariables]
+        public string _soundCollectionName = "dice";
+        [ViewVariables]
+        public int Step => _step;
+        [ViewVariables]
+        public int Sides => _sides;
+        [ViewVariables]
+        public int CurrentSide => _currentSide;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _step, "step", 1);
+            serializer.DataField(ref _sides, "sides", 20);
+            serializer.DataField(ref _soundCollectionName, "diceSoundCollection", "dice");
+            _currentSide = _sides;
+        }
+
+        public override void OnAdd()
+        {
+            base.OnAdd();
+            _random = new Random(Owner.Uid.GetHashCode() ^ DateTime.Now.GetHashCode();
```

Comment:
You're missing a ) here.
```suggestion
            _random = new Random(Owner.Uid.GetHashCode() ^ DateTime.Now.GetHashCode());
```